### ISSUE TITLE
3d_fullres inference generates artifacts

### ISSUE THREAD

**Issue by vieiraa** [MIC-DKFZ/nnUNet#1172]
Hello.
I'm training the nnUNet on a custom dataset of breast CTs. The dataset contain volumes with the same size in both width and height, but have different number of slices (varies from 286 to 800 slices). 
When I run inference, it tends to generate artifacts on volumes with a bigger number of slices (doesn't seem to affect the smaller ones).
For comparison, I trained with the 2d U-Net and it didn't generate these artifacts. Same thing if I change all phantoms to have the same number of slices.

Here we have an example of the inference (note the artifact on the bottom left):
![image](https://user-images.githubusercontent.com/19982663/191652058-fd83b321-8435-4726-933a-edcc139f928e.png)

And here we have the ground-truth:
![image](https://user-images.githubusercontent.com/19982663/191652198-03d43e63-4601-4749-b4f4-ddc9a59ee5aa.png)

Wonder if you may have an ideia on why this is happening, and if so, how to fix it. Been trying to figure this out for weeks.

Thank you.

**Comment by FabianIsensee** [comment]
I am assuming the phantom always has the same physical size? If that is the case, do you account for the change in voxel spacing when increasing/decreasing the number of slices? If you don't then nnU-Net might not knot what it is seeing. This especially happens if it is expected to predict all-background patches without ever seeing them during training.
Another thing you could try is the trainer class that uses BatchNorm instead if InstanceNorm. This could solve this issue as well

**Comment by vieiraa** [comment]
Using BatchNorm instead of InstanceNorm solved the issue. Thank you very much!!
